usdQt/opinionStackModel.py

```diff
@@ -0,0 +1,320 @@
+#
+# You may obtain a copy of the Apache License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the Apache License with the above modification is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the Apache License for the specific
+# language governing permissions and limitations under the Apache License.
+#
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import os.path
+
+from ._Qt import QtCore
+from pxr import Usd, Tf
+
+
+class _AttributeHandler(object):
+
+    def __init__(self, attributeName, timeCode):
+        self.attributeName = attributeName
+        self.timeCode = timeCode
+
+    def IsSpecified(self, primSpec):
+        if self.attributeName in primSpec.attributes:
+            if self.timeCode != Usd.TimeCode.Default() and \
+                    primSpec.attributes[self.attributeName].HasInfo('timeSamples'):
+                return True
+            return primSpec.attributes[self.attributeName].HasInfo('default')
+        return False
+
+    def GetValue(self, primSpec):
+        if primSpec.attributes[self.attributeName].HasInfo('default'):
+            return str(primSpec.attributes[self.attributeName].default)
+        elif primSpec.attributes[self.attributeName].HasInfo('timeSamples'):
+            return "TODO!"
+        else:
+            return None
+
+
+class _PrimMetadataHandler(object):
+
+    def __init__(self, metadataName):
+        self.metadataName = metadataName
+
+    def IsSpecified(self, primSpec):
+        return primSpec.HasInfo(self.metadataName)
+
+    def GetValue(self, primSpec):
+        if primSpec.HasInfo(self.metadataName):
+            return primSpec.GetInfo(self.metadataName)
+        return None
+
+
+class _PropertyMetadataHandler(object):
+
+    def __init__(self, propertyName, metadataName):
+        self.propertyName = propertyName
+        self.metadataName = metadataName
+
+    def IsSpecified(self, primSpec):
+        if self.propertyName in primSpec.properties:
+            return primSpec.properties[self.propertyName].HasInfo(self.metadataName)
+        return False
+
+    def GetValue(self, primSpec):
+        if primSpec.properties[self.propertyName]:
+            return primSpec.GetInfo(self.metadataName)
+        return None
+
+
+class _VariantSetsHandler(object):
+
+    def __init__(self):
+        pass
+
+    def IsSpecified(self, primSpec):
+        return bool(primSpec.variantSets)
+
+    def GetValue(self, primSpec):
+        return primSpec.variantSets
+
+
+class _VariantSetHandler(object):
+
+    def __init__(self, variantSet):
+        self.variantSet = variantSet
+
+    def IsSpecified(self, primSpec):
+        return self.variantSet in primSpec.variantSelections
+
+    def GetValue(self, primSpec):
+        return primSpec.variantSelections[self.variantSets]
```

Comment:
The name is `self.variantSet`. no `s`.  Btw, I *highly recommend* using PyCharm.  I'm basically just transcribing errors that it detected.  Also, I have some stub files for usd that will provide even better feedback when using the usd python bindings.